src/EnergyPlus/ZoneEquipmentManager.cc

```diff
@@ -3171,7 +3169,13 @@ void SimZoneEquipment(EnergyPlusData &state, bool const FirstHVACIteration, bool
 
             ZoneEquipTypeNum = state.dataZoneEquipmentManager->PrioritySimOrder(EquipTypeNum).EquipTypeEnum;
 
-            ZoneCompNum = state.dataZoneEquip->ZoneEquipList(state.dataSize->CurZoneEqNum).EquipIndex(EquipPtr);
+            // TODO: need a way to set ParentIndex for other than UnitarySystem so ZoneCompNum is correct when using ParentIndex
+            if (ZoneEquipTypeNum == ZoneEquip::PkgTermACAirToAir || ZoneEquipTypeNum == ZoneEquip::PkgTermHPAirToAir ||
+                ZoneEquipTypeNum == ZoneEquip::PkgTermHPWaterToAir) {
+                ZoneCompNum = state.dataZoneEquip->ZoneEquipList(state.dataSize->CurZoneEqNum).ParentIndex(EquipPtr);
+            } else {
+                ZoneCompNum = state.dataZoneEquip->ZoneEquipList(state.dataSize->CurZoneEqNum).EquipIndex(EquipPtr);
```

Comment:
For the unitary based systems, I don't think `EquipIndex` ever gets used so it can hold the value "parentindex" value and avoid this entire `if/else`.

Reply:
Maybe, I recall this caused an issue for some reason, just can't recall what right now. So I just did this the same way PTUnit did it. I can come back and clean all this up once this branch gets merged. I intend to do a lot of clean up in UnitarySystem as soon as possible. This is in SimPackagedTerminalUnit:

            if (SELECT_CASE_var == PkgTermHPAirToAir_Num) {
                PTUnitNum = CompIndex;
            } else if (SELECT_CASE_var == PkgTermACAirToAir_Num) {
                PTUnitNum = CompIndex + state.dataPTHP->NumPTHP;
            } else if (SELECT_CASE_var == PkgTermHPWaterToAir_Num) {
                PTUnitNum = CompIndex + state.dataPTHP->NumPTHP + state.dataPTHP->NumPTAC;
            } else {
                assert(false);
            }

Reply:
@rraustad Are you willing to try just one small change and use `EquipIndex` to hold this index (instead of putting it in `ParentIndex`) and blow up this if/else? Maybe it will help the 1.5% performance slowdown.

Reply:
Yes, cause I am curious why this was needed. I was just thinking that I could revert this and see what happens. It was something to do with each object type needing a 1-n index. SystemReports? I just can't recall. Go ahead and finish the code review so we can document all the changes needed and then open a follow up issue to clean those up.

Reply:
It's the zone availability manager that requires 1-n. So now the question is whether the new ParentIndex variable is needed.

    if (this->m_ZoneCompFlag) {
        state.dataHVACGlobal->ZoneComp(thisObjectType).ZoneCompAvailMgrs(this->m_EquipCompNum).AvailManagerListName =
            this->m_AvailManagerListName;
        state.dataHVACGlobal->ZoneComp(thisObjectType).ZoneCompAvailMgrs(this->m_EquipCompNum).ZoneNum = this->ControlZoneNum;
        this->m_ZoneCompFlag = false;
    }
    this->m_AvailStatus = state.dataHVACGlobal->ZoneComp(thisObjectType).ZoneCompAvailMgrs(this->m_EquipCompNum).AvailStatus;


    void UnitarySys::getPackagedTerminalUnitData
        if (sysNum == -1) {
            switch (getPTUnitType) {
            case 1:
                ++numPTAC;
                thisSys.m_EquipCompNum = numPTAC;
                break;
            case 2:
                ++numPTHP;
                thisSys.m_EquipCompNum = numPTHP;
                break;
            case 3:
                ++numPTWSHP;
                thisSys.m_EquipCompNum = numPTWSHP;
                break;
            default:
                assert(true);
            }